### PR TITLE
api: unify controllers

### LINKED ISSUE #415: controllers: unify them
Today we have three controllers:
- `UserController`
- `SystemController`
- `InfraController`

It doesn't make much sense to separate them since the API surface is now more coherent in the short-term plan of having only REST APIs. 

We should unify them.

### PR DESCRIPTION
**Pull request by jsign**
# Summary

Closes #415 

**Note: this PR is ready for review, but I'm leaving it as Draft until Bruno comes back for review***

# Context

See referenced issue.

# Implementation overview

Nothing entirely interesting since this is mostly refactoring simplifications.

# Implementation details and review orientation

NA

# Checklist
- [X]  Are changes backward compatible with existing SDKs, or is there a plan to manage it correctly?
- [X]  Are changes covered by existing tests, or were new tests included?
- [X]  Are code changes optimized for future code readers, commenting on problematic areas to understand (if any)?
- [X]  Future-self question: Did you avoid unjustified/unnecessary complexity to achieve the goal?

### CODE REVIEW

File: internal/router/controllers/controller.go [hunk 1]
```diff
@@ -24,14 +29,18 @@ type SQLRunner interface {
...
 }
 
-// UserController defines the HTTP handlers for interacting with user tables.
-type UserController struct {
-	runner SQLRunner
+// Controller defines the HTTP handlers for interacting with user tables.
+type Controller struct {
+	runner        SQLRunner
+	systemService system.SystemService
 }
```
Comment (jsign) [inline internal/router/controllers/controller.go:36]: Previous `UserController`, `SystemController` and `InfraController` where unified in `Controller`.

File: internal/router/controllers/controller.go [hunk 2]
```diff
@@ -210,9 +219,296 @@ func (c *UserController) GetTableRow(rw http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// Version returns git information of the running binary.
```
Comment (jsign) [inline internal/router/controllers/controller.go:222]: Moving methods around, nothing interesting.

File: internal/router/controllers/controller_test.go [hunk 3]
```diff
@@ -10,20 +11,21 @@ import (
 	"github.com/gorilla/mux"
 	"github.com/stretchr/testify/mock"
 	"github.com/stretchr/testify/require"
+	systemimpl "github.com/textileio/go-tableland/internal/system/impl"
 	"github.com/textileio/go-tableland/internal/tableland"
 	"github.com/textileio/go-tableland/mocks"
 )
 
-func TestUserController(t *testing.T) {
+func TestGetTableRow(t *testing.T) {
```
Comment (jsign) [inline internal/router/controllers/controller_test.go:19]: Improved namings a bit more.

File: internal/router/controllers/controller_test.go [hunk 4]
```diff
@@ -280,6 +282,165 @@ func TestUserControllerQueryExtracted(t *testing.T) {
 	}
 }
 
+func TestGetTablesByMocked(t *testing.T) {
```
Comment (jsign) [inline internal/router/controllers/controller_test.go:285]: Moved.

**Review by brunocalza — APPROVED** [review]
💯

**Pull request merged**